bin/transform.py

```diff
@@ -144,6 +149,63 @@ def lock(metadata):
 
     return metadata_readable, metadata_irida
 
+def categorize(metadata):
+
+    needed_cols = [
+        "host_scientific_name", "host_common_name", "food_product",
+        "environmental_site", "environmental_material"
+    ]
+    missing_cols = [col for col in needed_cols if col not in metadata.columns]
+
+    if len(missing_cols)>0:
+        raise KeyError("Missing required fields: " + str(missing_cols))
+
+    metadata_readable = metadata.copy(deep=True)
+
+    metadata_readable["calc_source_type"] = "Unknown"
+
+    metadata_readable.loc[
+        metadata_readable["host_scientific_name"].str.contains("homo sapiens", case=False, na=False) |
+        metadata_readable["host_common_name"].str.contains("human", case=False, na=False),
+        "calc_source_type"
+    ] = "Human"
+
+    metadata_readable.loc[
+        (~isna(metadata_readable["host_common_name"]) | ~isna(metadata_readable["host_common_name"])) &
+        (metadata_readable["calc_source_type"] == "Unknown"),
+        "calc_source_type"
+    ] = "Animal"
```

Comment:
I wonder if these can be re-written to use if-elif-else more explicitly. I see the `metadata_readable["calc_source_type"] == "Unknown"` part is meant to only work on something that hasn't already been modified, but I worried if someone comes back to this later to add or change the rules, there's an opportunity for missing that logic and causing overwrites when unintended.